rke/k8s_version_info.go

```diff
@@ -36,16 +36,17 @@ func loadRancherDefaultK8sVersions() map[string]string {
 		"2.6.3-patch1": "v1.21.x",
 		"2.6.3-patch2": "v1.21.x",
 		"2.6.4":        "v1.22.x",
+		"2.6.5":        "v1.23.x",
 		// rancher will use default if its version is absent
-		"default": "v1.22.x",
+		"default": "v1.23.x",
 	}
 }
 
 func loadRKEDefaultK8sVersions() map[string]string {
 	return map[string]string{
 		"0.3": "v1.16.3-rancher1-1",
 		// rke will use default if its version is absent
-		"default": "v1.22.9-rancher1-1",
+		"default": "v1.23.5-rancher1-1",
```

Comment:
Is this safe to merge this change into `dev-v2.6` branch provided we release `release-v2.6` from it? The other changes should be fine I think, just concerned about this one.
@kinarashah

Reply:
@snasovich Since this issue is for v2.6.5 milestone, I was under the impression that we would hold off on merging this PR till April patches are released. I don't think we want to mark v1.23 as default here until the local cluster is updated in rancher's codebase with v1.23 (which would be released with v2.6.5)

Reply:
@kinarashah @snasovich 
The April patch was released. So this PR can be merged now. 
There is [another PR](https://github.com/rancher/rancher/pull/37453) for updating the local cluster k8s version in rancher, and I will make sure that PR is merged first.